javalib/src/main/scala/java/util/Arrays.scala

```diff
@@ -617,11 +616,11 @@ object Arrays {
   }
 
   @inline private def checkIndicesForCopyOfRange(
-      len: Int, start: Int, end: Int): Unit = {
+      a: Array[_], start: Int, end: Int): Unit = {
     if (start > end)
       throw new IllegalArgumentException("" + start + " > " + end)
-    SemanticsUtils.arrayIndexOutOfBoundsCheck(start < 0 || start > len,
-        new ArrayIndexOutOfBoundsException)
+
+    a(start) // bounds check
```

Comment:
This does not do the same thing as before if `start == a.length`; it now throws. I doubt this is correct.